### PR TITLE
[MRG] Spatial colors

### PR DESCRIPTION
**Pull request by jaeilepp**
Takeover https://github.com/mne-tools/mne-python/pull/2534.

The colors are now shifted with red in x-direction and with blue in y-direction.
screenshot_2015-11-24_10-23-13

### CODE REVIEW

**Comment by jona-sassenhagen** [comment]
I don't like the red/blue, bright/dark thing. Much less usable IMO. I think if we use that, we should at the very least give an option to hook in different color assignments.

**Comment by agramfort** [comment]
send a PR to @jaeilepp to fix colormap or just suggest a change so we
can compare.

**Comment by jona-sassenhagen** [comment]
See PR for factoring out color assignments.

With the PR,
`plot_evoked(rerps["response"], spatial_colors=True);`
unknown-41
`plot_evoked(rerps["response"], spatial_colors="rgb");`
unknown-40

(Green lines are eye channels ... this is blinking around button press.)

**Comment by jona-sassenhagen** [comment]
I have code for the legend (topomap with colored sensors), but I think it might be cleaner to first merge this and add in the legend later.

But basically, it is:
```
def _plot_legend():
...
(draw empty head)
...
for x, y, c in zip(pos_x, pos_y, chspace):
        ax.add_artist(patches.Circle(xy=(x, y), radius=0.03, color=c))
```

and then in plot_evoked:
```
from mpl_toolkits.axes_grid.inset_locator import inset_axes
inset_axes = inset_axes(axis_for_timeseries, width=0.5, height=0.5, loc=3)
plot_legend(evoked, axis=inset_axes)
```

**Comment by jona-sassenhagen** [comment]
@kingjr you also have a nice color assignment scheme, although if you're busy, I can add try adding them to my PR myself.

**Comment by jaeilepp** [comment]
Now it seems to work pretty nicely with xyz. I guess earlier there was some kind of bug.
screenshot_2015-11-24_14-54-45
It looks so much better, that I'd remove the earlier 2d version and have only this.

**Comment by jona-sassenhagen** [comment]
The Argument for the other was that it's more directly interpretable. Maybe change the default to the rgb method, but keep this one.

**Comment by agramfort** [comment]
I like it. nice @jaeilepp !

**Comment by jaeilepp** [comment]
I added the rgb version to one of the examples. Are we ok at having a boolean for using 2d coloring scheme and a string for rgb? Perhaps a keyword for both color schemes would be better...

**Comment by agramfort** [comment]
I like the rgb mode rendering. Why not using only this one? the name rgb is
also not super meaningful. it's not a colormap name and it does not reflect
the xyz idea.

**Comment by jaeilepp** [comment]
Yeah. I'm fine with dropping the 2d version.

**Comment by jona-sassenhagen** [comment]
For an example, we can just take the rERP example and plot it with spatial_colors as true. Or some other example. don't need a dedicated example I hope. 

I can make a pr for the legend once this one is merged?

File: mne/viz/evoked.py [hunk 1]
```diff
@@ -236,12 +247,20 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
...
+                    else:
+                        colors = ['k'] * len(idx)
+                        for i in bad_ch_idx:
+                            if i in idx:
+                                colors[idx.index(i)] = 'r'
+                    for ch_idx in range(len(D)):
+                        line_list.append(ax.plot(times, D[ch_idx], picker=3.,
+                                                 zorder=0,
+                                                 color=colors[ch_idx])[0])
                 if gfp:  # 'only' or boolean True
```
Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: Maybe we need a different gfp color for `spatial_colors=True`, e.g. black.

File: mne/viz/evoked.py [hunk 2]
```diff
@@ -236,12 +247,20 @@ def _plot_evoked(evoked, picks, exclude, unit, show,
...
                     gfp_color = (0., 1., 0.)
```
Comment (jona-sassenhagen) [inline mne/viz/evoked.py]: E.g.
`gfp_color = ((1., 1., 1.) if spatial_colors else (0., 1., 0.))`

Comment (agramfort) [inline mne/viz/evoked.py]: +1

no need to add a parameter for this.

File: mne/viz/tests/test_evoked.py [hunk 3]
```diff
@@ -92,6 +92,8 @@ def test_plot_evoked():
         # plot with bad channels excluded
         evoked.plot(exclude='bads')
         evoked.plot(exclude=evoked.info['bads'])  # does the same thing
+        # spatial colors test
+        evoked.plot(spatial_colors=True)
```
Comment (agramfort) [inline mne/viz/tests/test_evoked.py]: maybe update an existing line? we're fighting test time as much as possible

**Comment by agramfort** [comment]
I am on it. I'll merge by rebase.

**Comment by agramfort** [comment]
merged by rebase. Thx @jaeilepp

**Comment by jona-sassenhagen** [comment]
Nice!

**Pull request closed without merging**